### PR TITLE
omaha: Make the trivial server to only return newer packages.

### PR DESCRIPTION
**Pull request by yifan-gu**
This is for adding related test in TCO.

### CODE REVIEW

File: omaha/trivial_server_test.go [hunk 1]
```diff
@@ -82,6 +83,37 @@ func TestTrivialServer(t *testing.T) {
 		t.Fatalf("failed to parse body: %v", err)
 	}
 
+	// Should get zero update because the version is already the latest.
+	if len(resp.Apps) != 1 ||
```
Comment (lucab) [inline omaha/trivial_server_test.go]: Is this `if` missing the body? Or is is implicitly trying to panic on wrong responses?

Comment (yifan-gu) [inline omaha/trivial_server_test.go]: Ehh, was debugging this before, forgot to add back. Good catch!

Comment (yifan-gu) [inline omaha/trivial_server_test.go]: @lucab Fixed

**Comment by yifan-gu** [comment]
@lucab Addressed the comment, PTAL :)

File: omaha/trivial_server_test.go (hunk 1 above)
Comment (lucab) [inline omaha/trivial_server_test.go]: (I think you forgot to push)

**Comment by lucab** [comment]
To avoid another round-trip delay, everything else except the pending test fix LGTM.

**Pull request merged**